src/ServerScriptService/Services/WeldService.luau

```diff
@@ -0,0 +1,68 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Graph = require(ReplicatedStorage.Common.Graph)
+local Knit = require(ReplicatedStorage.Packages.Knit)
+
+local WeldService = Knit.CreateService {
+    Name = "WeldService";
+    Client = {};
+}
+
+function WeldService:WeldModel(model)
+    local primaryPart = assert(model.PrimaryPart, "Model has no primary part!")
+
+    for _, v in model:GetChildren() do
+        if v == model.PrimaryPart then
+            continue
+        end
+
+        local weld = Instance.new("WeldConstraint")
+        weld.Part0 = primaryPart
+        weld.Part1 = v
+        weld.Parent = model
+    end
+end
+
+function WeldService:RemoveModel(model)
+    for otherModel, weld in self._weldGraph:GetEdges(model) do
+        weld:Destroy()
+        self._weldGraph:Disconnect(model, otherModel)
+    end
+
+    self._connections[model]:Disconnect()
+    self._connections[model] = nil
```

Comment:
Let's check if there is a connection first, before destroying and nilling, to ensure the function is idempotent and doesn't error after multiple calls.